Extensibility - show connected resources (#1495)
kyma-project/busola: docs/extensibility/resources.md

]
```

## relations section

> This section contains an object that maps a relation name to relation configuration object. Relation name preceded by dollar sign '\$' can be then used in `path` expression.

Reviewer: ```suggestion
The `relations` section contains an object that maps a relation name to a relation configuration object. The relation name preceded by a dollar sign '\$' can be then used in the `path` expression.
```

"Can" it be used, or "is" it used? Does "can" imply it's optional or there are other alternatives? Maybe we can drop the "can".

Reviewer: Alright! Thank you for clarification :)

> It's possible to use both relation name and a path, e.g. `{"path": $myRelatedResource.metadata.labels}` will return the `metadata.labels` of related resource.

Reviewer: ```suggestion
It's possible to use both relation name and a path; for example, `{"path": $myRelatedResource.metadata.labels}` returns the `metadata.labels` of the related resource.
```

### Relation configuration object fields

Those fields are used to build the related resource URL and filter the received data.

- **kind** - _[required]_ Kubernetes resource kind.
- **group** - _[required]_ Kubernetes resource group.
- **version** - _[required]_ Kubernetes resource version.

> - **namespace** - resource's Namespace name, defaults to original resource's Namespace. If set to `null` the relation with match clusterwide resources / resources in all namespaces.

Reviewer: ```suggestion
- **namespace** - the resource's Namespace name; it defaults to the original resource's Namespace. If set to `null`, the relation matches cluster-wide resources or resources in all Namespaces.
```

I'm guessing "with" is a typo for "will", and I'd say we can replace future tense with present tense here.

> - **resourceName** - specific resource name, leave empty to match all resources of given type.

Reviewer: ```suggestion
- **resourceName** - a specific resource name; leave empty to match all resources of a given type.
```

> - **ownerLabelSelectorPath** - path to original object's `selector` type property (e.g. `spec.selector.matchLabels` for Deployment, used to select matching Pods).

Reviewer: ```suggestion
- **ownerLabelSelectorPath** - the path to original object's `selector` type property; for example, `spec.selector.matchLabels` for Deployment, used to select matching Pods.
```

> - **selector** - [JSONata](https://docs.jsonata.org/overview.html) function enabling user to write a custom matching logic. It receives a data context of:

Reviewer: ```suggestion
- **selector** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It receives a data context of:
```

```

If you provide both `translations` and `translations-{lang}` sections, they are merged together.

### Value preprocessors

> Value preprocessors are used as a middleware between a value and the actual renderer. They can transform a given value and pass it further or stop processing, rendering to view early.

Reviewer: ```suggestion
Value preprocessors are used as a middleware between a value and the actual renderer. They can transform a given value and pass it to the widget; or stop processing and render it so you can view it immediately without passing it to the widget.
```

#### List of value preprocessors

- **PendingWrapper** - useful when value resolves to a triple of `{loading, error, data}`:

>   - For `loading` equal to `true`, it will display a loading indicator.

Reviewer: ```suggestion
  - For `loading` equal to `true`, it displays a loading indicator.
```

>   - For truthy `error`, it will display an error message.

Reviewer: ```suggestion
  - For truthy `error`, it displays an error message.
```

>   - Otherwise, it will pass `data` to display component.

Reviewer: ```suggestion
  - Otherwise, it passes `data` to the display component.
```

>   Unless custom handling of error or loading state is required, using the `PendingWrapper` is recommended, e.g. for fields that use [related resources](#relations-section).

Reviewer: ```suggestion
  Unless you need custom handling of error or loading state, we recommend using the `PendingWrapper`, for example, for fields that use [related resources](#relations-section).
```
